public/app/features/plugins/extensions/registry/ExposedComponentsRegistry.ts

```diff
@@ -0,0 +1,72 @@
+import { PluginExposedComponentConfig } from '@grafana/data';
+
+import { PluginPreloadResult } from '../../pluginPreloader';
+import { logWarning } from '../utils';
+
+import { Registry } from './Registry';
+
+export type RegistryType = {
+  [id: string]: {
+    pluginId: string;
+    config: PluginExposedComponentConfig;
+  };
+};
+
+export class ExposedComponentsRegistry extends Registry<RegistryType> {
+  constructor(initialState: RegistryType = {}) {
+    super({
+      initialState,
+    });
+  }
+
+  mapToRegistry(registry: RegistryType, item: PluginPreloadResult): RegistryType {
```

Comment:
What if instead of having a "common" type for the different kind of preload results, this method would just receive a list of specific `PluginExposedComponentConfig[]`?  
```suggestion
  mapToRegistry(registry: RegistryType, pluginId: string, configs: PluginExposedComponentConfig[]): RegistryType {
```